### ISSUE TITLE
Wrong Testing Result Display

### ISSUE THREAD

**Issue by Ao-Last** [microsoft/vscode-java-test#1659]
# What happened? 

Run test from the side panel and it display with green icon while the test cannot even build.

<img width="1825" alt="image" src="https://github.com/microsoft/vscode-java-test/assets/63785048/8a6d5955-a176-4853-ad06-e8d8b5f38f22">

**Comment by jdneo** [comment]
Those 'dark' green check marks are the history test status.

You can run the command `Test: Clear All Results` to reset them.

What's your build error in the output channel?

**Comment by Ao-Last** [comment]
> Those 'dark' green check marks are the history test status.
> 
> You can run the command `Test: Clear All Results` to reset them.
> 
> What's your build error in the output channel?

Thanks for pointing out the 'dark' icon's meaning. The build error is caused by an undefined symbol. I cannot reproduce it again and I will reopen the issue once I meet it again.

**Comment by Ao-Last** [comment]
To reproduce,

1. make an build error caused by a wrong usage of assert methods. E.g. assertTrue(String)
<img width="1855" alt="image" src="https://github.com/microsoft/vscode-java-test/assets/63785048/c9f854b9-1d5d-4f5b-aca7-cbd5b7e275a6">
2. run the test from the sidebar by clicking the triangle button and wait for prompt error message
<img width="1877" alt="image" src="https://github.com/microsoft/vscode-java-test/assets/63785048/18840df6-72f9-4582-ac7c-13711235871d">
3. click "Continue", and the icon would be green check again.
<img width="1840" alt="image" src="https://github.com/microsoft/vscode-java-test/assets/63785048/25ffe57a-1ee7-417b-af82-25ece48c3f4c">

Build error:
<img width="1920" alt="image" src="https://github.com/microsoft/vscode-java-test/assets/63785048/2837651d-6129-4361-b899-073fa8a3234e">

**Comment by jdneo** [comment]
Ok, so the green check in step 3 is a history status.

A better solution might be that display the compilation error in the peek view and mark the test as failed.

**Comment by Ao-Last** [comment]
Yep, it shows the history status when meeting a compilation error. I suggest to mark it as a test error instead of test failed to distinguish between wrong test code and wrong source code.
